lib/tests/streamlit/user_info_test.py

```diff
@@ -118,3 +142,81 @@ def test_st_user_reads_from_context_(self):
             raise e
         finally:
             add_script_run_ctx(threading.current_thread(), orig_report_ctx)
+
+
+@patch(
+    "streamlit.user_info.secrets_singleton",
+    MagicMock(
+        load_if_toml_exists=MagicMock(return_value=True),
+        get=MagicMock(return_value=SECRETS_MOCK),
+    ),
+)
+class UserInfoAuthTest(DeltaGeneratorTestCase):
+    """Test UserInfoProxy Auth functionality."""
+
+    @parameterized.expand(["google", "microsoft", "auth0"])
+    def test_user_login(self, provider):
+        """Test that st.experimental_user.login sends correct proto message."""
+        st.experimental_user.login(provider)
+
+        c = self.get_message_from_queue().auth_redirect
+
+        assert c.url.startswith("/auth/login?provider=")
+
+        jwt_token = c.url.split("=")[1]
+        raw_payload = jwt_token.split(".")[1]
+        parsed_payload = json.loads(base64.b64decode(raw_payload + "==="))
+
+        assert parsed_payload["provider"] == provider
+
+    def test_user_login_with_invalid_provider(self):
+        """Test that st.experimental_user.login sends correct proto message."""
+        with self.assertRaises(StreamlitAPIException) as ex:
+            st.experimental_user.login("invalid_provider")
+
+        assert "Auth credentials are missing *'invalid_provider'*" in str(ex.exception)
+
+    def test_user_login_redirect_uri_missing(self):
```

Comment:
I'd recommend having a description for each test of what to expect. Something like "Tests that an error is raised if the redirect uri is missing"

I didn't check all of the tests but just make sure the relevant info is there.

Reply:
done